src/mame/mr/nl_dribling.cpp

```diff
@@ -0,0 +1,597 @@
+	NET_C(R54.1, GND)
+	NET_C(R53.2, C31.1)
+	NET_C(C31.2, R48.1, R56.2, Q9.B)
+	NET_C(R48.2, I_V5)
+	NET_C(R56.1, GND)
+	NET_C(Q9.C, R55.1)
+	ALIAS(OUTPUT, Q9.C)
+	NET_C(R55.2, I_V5)
+	NET_C(Q9.E, R57.2, C32.2)
+	NET_C(R57.1, GND)
+	NET_C(C32.1, R58.2)
+	NET_C(R58.1, GND)
+}
+
+static NETLIST_START(stop_palla)
+{
+	ANALOG_INPUT(I_V5, 5)
+	QBJT_EB(Q1, "BC309")
+	RES(R5, RES_K(1))
+	RES(R6, RES_K(160))
+	RES(R7, RES_K(150))
+	RES(R8, RES_K(10))
+	RES(R9, RES_K(2.2))
+	CAP(C5, CAP_U(1))
+	CAP(C6, CAP_U(1))
+	DIODE(D1, "1N914")
+	SUBMODEL(LM339, IC_B9)
+	ALIAS(GND, IC_B9.4)
+	ALIAS(INPUT, R5.1)
+	NET_C(R5.2, Q1.B)
+	NET_C(Q1.E, I_V5)
+	NET_C(Q1.C, C5.2, R7.2, R6.1)
+	NET_C(C5.1, GND)
+	NET_C(R7.1, GND)
+	NET_C(R6.2, IC_B9.1, D1.A)
+	NET_C(D1.K, R9.1, IC_B9.5, R8.2)
+	NET_C(IC_B9.2, C6.2, R8.1)
+	NET_C(C6.1, GND)
+	NET_C(R9.2, I_V5)
+	NET_C(IC_B9.3, I_V5)
+
+	ALIAS(OUTPUT, IC_B9.5)
+}
+
+// The actual PARATA schematics requires JFETs (2N3819) and using 'MOSFET(Q21, "NMOS(VTO=-1.0)")' as proposed in the FAQ 
+// doesn't work. So using the same circuit STOP_PALLA with different values to emulate the sound which has a higher pitch.
+#ifdef USE_FAKE_PARATA
+static NETLIST_START(parata)
+{
+	ANALOG_INPUT(I_V5, 5)
+	QBJT_EB(Q1, "BC309")
+	RES(R5, RES_K(1))
+	RES(R6, RES_K(160))
+	RES(R7, RES_K(75))
+	RES(R8, RES_K(3.75))
+	RES(R9, RES_K(2.2))
+	CAP(C5, CAP_U(1))
+	CAP(C6, CAP_U(1))
+	DIODE(D1, "1N914")
+	SUBMODEL(LM339, IC_B9)
+	ALIAS(GND, IC_B9.4)
+	ALIAS(INPUT, R5.1)
+	NET_C(R5.2, Q1.B)
+	NET_C(Q1.E, I_V5)
+	NET_C(Q1.C, C5.2, R7.2, R6.1)
+	NET_C(C5.1, GND)
+	NET_C(R7.1, GND)
+	NET_C(R6.2, IC_B9.1, D1.A)
+	NET_C(D1.K, R9.1, IC_B9.5, R8.2)
+	NET_C(IC_B9.2, C6.2, R8.1)
+	NET_C(C6.1, GND)
+	NET_C(R9.2, I_V5)
+	NET_C(IC_B9.3, I_V5)
+
+	ALIAS(OUTPUT, IC_B9.5)
+}
+#else
+static NETLIST_START(parata)
+{
+	ANALOG_INPUT(I_V5, 5)
+	QBJT_EB(Q1, "BC309")
+	Q_2N3819(Q2)
+	Q_2N3819(Q3)
+	RES(R5, RES_K(1))
+	RES(R6, RES_K(150))
+	RES(R7, RES_K(220))
+	RES(R8, RES_K(1))
+	RES(R9, RES_K(2.2))
+	CAP(C5, CAP_U(2.2))
+	CAP(C6, CAP_U(1))
+	DIODE(D1, "1N914")
+	SUBMODEL(LM339, IC_B9)
+	ALIAS(GND, IC_B9.4)
+	ALIAS(INPUT, R5.1)
+	NET_C(R5.2, Q1.B)
+	NET_C(Q1.E, I_V5)
+	NET_C(Q1.C, C5.2, R7.2, R6.1)
+	NET_C(C5.1, GND)
+	NET_C(R7.1, GND)
+	NET_C(R6.2, IC_B9.1, D1.A)
+	NET_C(D1.K, R9.1, IC_B9.5, Q2.G, Q2.S)
+	NET_C(IC_B9.2, C6.2, Q2.D, Q3.G)
+	NET_C(C6.1, GND)
+	NET_C(R9.2, Q3.S, I_V5)
+	NET_C(IC_B9.3, I_V5)
+	NET_C(Q3.D, R8.2)
+	NET_C(R8.1, GND)
+	ALIAS(OUTPUT, Q3.D)
+}
+#endif
+
+// Third order Butterworth filter with 15KHz cutoff frequency.
+// Values computed using https://www.electronicproducts.com/tools/op-amp-low-pass-butterworth-filter
+// and then adjusted to remove the ringing noise.
+// This is because the tone generator outputs 40KHz at idle, and this is to avoid aliasing when outputing at 48KHz.
+static NETLIST_START(output_filter)
+{
+    // 15KHz low-pass filter
+    OPAMP(AMP, "OPAMP(TYPE=1 FPF=5 RI=1M RO=50 UGF=1M SLEW=1M VLH=0.5 VLL=0.03 DAB=0.0015)")
```

Comment:
Are all the parameters necessary?  I though the type 1 idealised op-amp model didn’t use SLEW, VLH, VLL or DAB.  Those are only used by the more detailed type 3 model.

See the table here: https://github.com/mamedev/mame/blob/mame0263/src/lib/netlist/analog/nld_opamps.cpp#L37

The first column tells you which models use the parameters.

Reply:
> I had to populate more parameters, but I used the ones from the example and it sounds well. But I am no expert.

I guess it always checks that they’re present, but just doesn’t use them for the type 1 model.

Reply:
They are necessary even if they are not used. It throws a runtime error if they are missing. That's how the models work